cpp/src/arrow/compute/kernels/scalar_validity.cc

```diff
@@ -202,23 +221,29 @@ const FunctionDoc is_inf_doc(
     ("For each input value, emit true iff the value is infinite (inf or -inf)."),
     {"values"});
 
-const FunctionDoc is_null_doc("Return true if null",
-                              ("For each input value, emit true iff the value is null."),
-                              {"values"});
+const FunctionDoc is_null_doc(
+    "Return true if null, NaN can be considered as null",
+    ("For each input value, emit true iff the value is null. Default behavior is to emit "
+     "false for NaN values. True can be emitted for NaN values by toggling "
+     "NanNullOptions flag."),
+    {"values"}, "NanNullOptions");
 
 const FunctionDoc is_nan_doc("Return true if NaN",
                              ("For each input value, emit true iff the value is NaN."),
                              {"values"});
 
 }  // namespace
 
+FunctionOptions kNanNullOptions = NanNullOptions::Defaults();
```

Comment:
You can put this in the anonymous namespace above.

Reply:
It was addressed on 2e772c1